src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.edit-telephone.unit.spec.jsx

```diff
@@ -0,0 +1,238 @@
+import React from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { waitForElementToBeRemoved } from '@testing-library/react';
+import user from '@testing-library/user-event';
+import { expect } from 'chai';
+import { setupServer } from 'msw/node';
+
+import { resetFetch } from 'platform/testing/unit/helpers';
+import { FIELD_TITLES, FIELD_NAMES } from 'vet360/constants';
+
+import * as mocks from '../../../msw-mocks';
+import PersonalInformation from '../../../components/personal-information/PersonalInformation';
+
+import {
+  createBasicInitialState,
+  renderWithProfileReducers,
+} from '../../unit-test-helpers';
+import { beforeEach } from 'mocha';
+
+const newAreaCode = '415';
+const newPhoneNumber = '555-0055';
+const ui = (
+  <MemoryRouter>
+    <PersonalInformation />
+  </MemoryRouter>
+);
+let view;
+let server;
+
+function getEditButton(numberName) {
+  let editButton = view.queryByText(new RegExp(`add.*${numberName}`, 'i'), {
+    selector: 'button',
+  });
+  if (!editButton) {
+    // Need to use `queryByRole` since the visible label is simply `Edit`, but
+    // the aria-label is more descriptive
+    editButton = view.queryByRole('button', {
+      name: new RegExp(`edit.*${numberName}`, 'i'),
+    });
+  }
+  return editButton;
+}
+
+// helper function that enters the `Edit phone number` view, enters a number,
+// and clicks the `Update` button.
+function editPhoneNumber(numberName) {
+  const editButton = getEditButton(numberName);
+  editButton.click();
+
+  const phoneNumberInput = view.getByLabelText(/Number/);
+  const extensionInput = view.getByLabelText(/Extension/);
+  expect(phoneNumberInput).to.exist;
+
+  // enter a new email address in the form
+  user.clear(phoneNumberInput);
+  user.type(phoneNumberInput, `${newAreaCode} ${newPhoneNumber}`);
+  user.clear(extensionInput);
+
+  // save
+  view.getByText('Update', { selector: 'button' }).click();
+
+  return { phoneNumberInput };
+}
+
+// When the update happens while the Edit View is still active
+async function testQuickSuccess(numberName) {
+  server.use(...mocks.transactionSucceeded);
+
+  const { phoneNumberInput } = editPhoneNumber(numberName);
+
+  // wait for the edit mode to exit
+  await waitForElementToBeRemoved(phoneNumberInput);
+
+  // the 'edit' button should exist
+  expect(
+    view.getByRole('button', { name: new RegExp(`edit.*${numberName}`, 'i') }),
+  ).to.exist;
+  // and the new number should exist in the DOM
+  // TODO: make better assertions for this?
+  expect(view.getAllByText(newAreaCode, { exact: false }).length).to.eql(4);
```

Comment:
Learned something new here, `exact: false` "matches substrings and is not case-sensitive" 👍

Reply:
Yeah, I think the alternative here would be something like `getAllByText(new RegExp(newAreaCode))` or maybe even `getAllByText(new RegExp(`${newAreaCode}`))`